pkg/agent/goclient_opensearch.go

```diff
@@ -0,0 +1,169 @@
+package agent
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/google/pprof/profile"
+	"github.com/joho/godotenv"
+	opensearch "github.com/opensearch-project/opensearch-go"
+	opensearchapi "github.com/opensearch-project/opensearch-go/opensearchapi"
+)
+
+var IsInsecure = false
+var GlobalStoreAddress = ""
+var TheClient *opensearch.Client
+
+const IndexName = "parca_agent_profile"
+
+type Client struct {
+	Name       string `json:"name"`
+	Value      string `json:"value"`
+	Periodtype string `json:"periodtype"`
+	Period     int64  `json:"period"`
+	Time       int64  `json:"time"`
+	Duration   string `json:"duration"`
+	Samples    string `json:"samples"`
+	Locations  string `json:"locations"`
+	Mappings   string `json:"mappings"`
+	Functions  string `json:"functions"`
+}
+
+func GoCreateClient(theStoreAddress string) {
+	//check if it starts with http or https and then append if not there
+	if theStoreAddress[0:1] == "h" {
+		GlobalStoreAddress = theStoreAddress
+	} else {
+		GlobalStoreAddress = "http://" + theStoreAddress
+	}
+
+	//Load the Username and Password
+	envErr := godotenv.Load(".env")
+	if envErr != nil {
+		log.Print("Could not load .env file\n")
+	}
+
+	//fmt.Printf("The address is %s\n", GlobalStoreAddress)
+	client, err := opensearch.NewClient(opensearch.Config{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: IsInsecure},
+		},
+		Addresses: []string{GlobalStoreAddress},
+		Username:  os.Getenv("OPENSEARCH_USERNAME"),
+		Password:  os.Getenv("OPENSEARCH_PASSWORD"),
+	})
+	//Make it a .env file located at top of directory
+	//OPENSEARCH_USERNAME=??
+	//OPENSEARCH_PASSWORD=??
+	//fmt.Printf("InSecure setto %s\n", IsInsecure)
+	//fmt.Printf("Opensearch Username is %s\n", os.Getenv("OPENSEARCH_USERNAME"))
+	//fmt.Printf("Opensearch Password is %s\n", os.Getenv("OPENSEARCH_PASSWORD"))
+	TheClient = client
```

Comment:
can this be assigned at line 55 instead of creating a temp var `client` then reassign here?